File: PathStructure.WatcherHost/Dtos/StatusNotificationParams.cs
namespace PathStructure.WatcherHost.Dtos
{
    /// <summary>
    /// Represents status notification parameters.
    /// </summary>
    internal sealed class StatusNotificationParams
    {
        /// <summary>
        /// Gets or sets the status message.
        /// </summary>
        public string Message { get; set; }

        /// <summary>
        /// Gets or sets the connection state string.
        /// </summary>
        public string State { get; set; }

        /// <summary>
        /// Gets or sets the ISO timestamp for the status event.
        /// </summary>
        public string Timestamp { get; set; }
    }
}

File: PathStructureClient/src/dtos/StatusNotificationParams.js
class StatusNotificationParams {
  /**
   * @param {object} options
   * @param {string} options.message
   * @param {string} options.state
   * @param {string} options.timestamp
   */
  constructor({ message, state, timestamp }) {
    /** @type {string} */
    this.message = message;
    /** @type {string} */
    this.state = state;
    /** @type {string} */
    this.timestamp = timestamp;
  }
}

module.exports = { StatusNotificationParams };
